src/pages/billing/List.js

```diff
@@ -360,12 +360,20 @@ export const BillingList = ({ queryParams: { selectedName } }) => {
               p(['It may not exist, or you may not have access to it.'])
             ])
           ])],
-        [selectedName && hasBillingProjects, () => h(ProjectDetail, {
-          key: selectedName,
-          project: _.find({ projectName: selectedName }, billingProjects),
-          billingAccounts,
-          authorizeAndLoadAccounts
-        })],
+        [selectedName && hasBillingProjects, () => {
+          const index = _.findIndex({ projectName: selectedName }, billingProjects)
+          return h(ProjectDetail, {
+            key: selectedName,
+            billingProject: billingProjects[index],
+            billingAccounts,
+            authorizeAndLoadAccounts,
+            updateProjectAndReload: async () => {
+              const projects = billingProjects.slice()
+              projects[index] = await Ajax(signal).Billing.billingProject(selectedName)
+              setBillingProjects(projects)
```

Comment:
1. Should the effect that watches for updated billing projects on 267 have a dependency on `billingProjects`, so that it's not re-evaluated on every render?
2. To make sure we're reading from the backend as the single source of truth (as opposed to modeling what we think it looks like locally), I'd recommend calling `loadProjects` here, instead of just loading the one. That would also let you get rid of the whole find-the-index pattern, and you could just use `_.find`.

Reply:
We're changing one billing project so reloading all projects is a bit unnecessary. I understand what you mean about using the backend as a single source of truth but I believe one must have some knowledge of what one's doing.